app/app/crud/crud_school.py

```diff
@@ -0,0 +1,22 @@
+from typing import List, Optional
+
+from sqlalchemy.orm import Session
+
+from app.crud.base import CRUDBase
+from app.models import User
+from app.models.school import School
+from app.schemas.school import SchoolCreate, SchoolUpdate
+
+
+class CRUDSchool(CRUDBase[School, SchoolCreate, SchoolUpdate]):
+    def get_by_name(self, db: Session, *, name: str) -> Optional[School]:
+        return db.query(School).filter(School.name == name).first()
+
+    def get_all_students(self, db: Session, *, school_id: int) -> List[User]:
```

Comment:
Wrap this and the next function's return type in an `Optional[]`? (Or List[Optional[User]], incase there's no matching students/professors.